### PR TITLE
Feature/6d key teleop

### LINKED ISSUE #6: [6D-key_teleop] initial teleoperation implementation
based on #1 

The virtual robot moves "somehow" via keyboard presses. This should implement / fork an initial keyboard control and show, what needs to be done in future PBIs.

Evaluate, if possible or already available
* robot stops on key-release
* single-joint jogging
* cartesian 6D jogging
* velocity scaling
* define other coordinate frame (like tcp cartesian jogging)

### PR DESCRIPTION
**Pull request by DamKoVosh**
I have a first version of the 6k key teleoperation package.
Please review the current implementation and test it.
Tests, Documentation and Instruction will follow soon
fixes #6

### CODE REVIEW

**Comment by DamKoVosh** [comment]
There are still some seconds until the robot stops, when pressing a key for a longer time. Not quire sure, where this comes from yet.

File: pilz_teleoperation/src/pilz_teleoperation/terminal_text_window.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        """
+        super(TerminalTextWindow, self).__init__(*args, **kwargs)
+        self._screen = stdscr
+        curses.curs_set(0)
+
+    def _redraw(self):
+        self._start_page()
+        self._write_line(1, "PILZ teleoperation driver")
+        self._write_line(3, "Settings:")
+        self._write_line(4, "  - linear velocity: %.2f%% \t\t angular velocity: %.2f%%"
```
Comment (Hardthof) [inline pilz_teleoperation/src/pilz_teleoperation/terminal_text_window.py]: float value is not percent

File: pilz_teleoperation/src/pilz_teleoperation/teleoperation_settings.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+
+class TeleoperationSettings:
+    setting_change_method_bindings = {
+        SetTeleopSettingsRequest.DECREASE_ANGULAR_VELOCITY: _decrease_angular_velocity,
+        SetTeleopSettingsRequest.INCREASE_ANGULAR_VELOCITY: _increase_angular_velocity,
+        SetTeleopSettingsRequest.DECREASE_LINEAR_VELOCITY: _decrease_linear_velocity,
+        SetTeleopSettingsRequest.INCREASE_LINEAR_VELOCITY: _increase_linear_velocity,
+        SetTeleopSettingsRequest.TOGGLE_WORLD_AND_TCP_FRAME: _toggle_target_frame_world_tcp,
+        SetTeleopSettingsRequest.TOGGLE_PLANE: _toggle_projection_plane
```
Comment (Hardthof) [inline pilz_teleoperation/src/pilz_teleoperation/teleoperation_settings.py:72]: toggle controller not implemented

File: pilz_teleoperation/src/pilz_teleoperation/teleoperation_driver.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    def norm(self):
+        t_sum = abs(self.linear.x) + abs(self.linear.y) + abs(self.linear.z)
+        if t_sum > 0:
+            self.linear.x /= t_sum
+            self.linear.y /= t_sum
+            self.linear.z /= t_sum
+
+    def scale_linear_velocity(self, lin_vel):
+        self.linear.x *= _teleop_settings.MAX_JOG_RANGE * lin_vel
```
Comment (Hardthof) [inline pilz_teleoperation/src/pilz_teleoperation/teleoperation_driver.py]: is max jog range necessary?

File: pilz_teleoperation/src/pilz_teleoperation/teleoperation_driver.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                               twist.angular.y,
+                               twist.angular.z)
+
+    def project_on_plane(self, projection_plane):
+        if projection_plane == SetTeleopSettingsRequest.USE_XZ_PLANE:
+            self.linear.y, self.linear.z = 0, self.linear.y
+        elif projection_plane == SetTeleopSettingsRequest.USE_YZ_PLANE:
+            self.linear.x, self.linear.y, self.linear.z = 0, self.linear.x, self.linear.y
+
+    def norm(self):
```
Comment (Hardthof) [inline pilz_teleoperation/src/pilz_teleoperation/teleoperation_driver.py]: is norm necessary?

File: pilz_teleoperation/src/pilz_teleoperation/teleoperation_settings.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
+from pilz_teleoperation.srv import SetTeleopSettingsRequest
```
Comment (Hardthof) [inline pilz_teleoperation/src/pilz_teleoperation/teleoperation_settings.py:16]: code docu

**Review by Hardthof — DISMISSED** [review]
- License Nodes in Files
- What level of code documentation for protected/private functions is required?

File: pilz_teleoperation/src/pilz_teleoperation/teleoperation_settings.py (hunk 2 above)
Comment (jschleicher) [reply pilz_teleoperation/src/pilz_teleoperation/teleoperation_settings.py:72]: not part of this initial implementation. Removed from screen output.

**Comment by jschleicher** [comment]
>     * What level of code documentation for protected/private functions is required?

Tiny improvements in documentation see c1b6662. Readme added. Deferred in-depth tests and more documentation for the next story, since major changes quite possible after sprint review.

**Review by jschleicher — APPROVED** [review]
:+1:

**Pull request merged**